File: compatibility/minification/src/main/java/com/lelloman/paravoidcompat/minification/ProbeProvider.java
package com.lelloman.paravoidcompat.minification;

import android.content.ContentProvider;
import android.content.ContentValues;
import android.database.Cursor;
import android.net.Uri;
import android.os.Bundle;

public final class ProbeProvider extends ContentProvider {
    @Override public boolean onCreate() { return true; }
    @Override public Bundle call(String method, String arg, Bundle extras) {
        Bundle result = new Bundle();
        result.putString("result", "ping".equals(method) ? "pong" : "unknown");
        return result;
    }
    @Override public Cursor query(Uri uri, String[] projection, String selection, String[] args, String order) { return null; }
    @Override public String getType(Uri uri) { return null; }
    @Override public Uri insert(Uri uri, ContentValues values) { return null; }
    @Override public int delete(Uri uri, String selection, String[] args) { return 0; }
    @Override public int update(Uri uri, ContentValues values, String selection, String[] args) { return 0; }
}

File: compatibility/language/src/main/java/com/lelloman/paravoidcompat/language/ProbeProvider.kt
package com.lelloman.paravoidcompat.language

import android.content.ContentProvider
import android.content.ContentValues
import android.database.Cursor
import android.net.Uri

class ProbeProvider : ContentProvider() {
    override fun onCreate(): Boolean {
        result = outcome {
            check(ProbeApplication.mainResult == "not run")
            check(Probes.discovery())
        }
        return true
    }
    override fun query(uri: Uri, projection: Array<out String>?, selection: String?, args: Array<out String>?, sort: String?): Cursor? = null
    override fun getType(uri: Uri): String? = null
    override fun insert(uri: Uri, values: ContentValues?): Uri? = null
    override fun delete(uri: Uri, selection: String?, args: Array<out String>?): Int = 0
    override fun update(uri: Uri, values: ContentValues?, selection: String?, args: Array<out String>?): Int = 0
    companion object { var result = "not run" }
}
